react/server.js

```diff
@@ -0,0 +1,256 @@
+      type: req.body.intervalsType,
+      start_date: req.body.intervalsStartDate,
+      end_date: req.body.intervalsEndDate,
+    });
+
+  // Reach out to the Pelm API intervals endpoint providing all the relevent headers
+  fetch(url, requestOptions)
+    .then((response) => {
+      if (response.ok) {
+        return response.json();
+      } else {
+        return response.text().then((text) => {
+          throw new Error(text);
+        });
+      }
+    })
+    .then((data) => {
+      // Return intervals data
+      res.end(JSON.stringify(data));
+    })
+    .catch((error) => {
+      try {
+        const errorObject = JSON.parse(error.message);
+        console.log(errorObject);
+      } catch (e) {
+        console.log('an error occurred');
+      }
+    });
+
+})
+
+
+// Get Bills
+// Link to Pelm docs: https://docs.pelm.com/reference/get_bills
+app.post('/bills', (req, res) => {
+
+  const accessToken = req.session.access_token
+  const headers = new Headers();
+  headers.set('Authorization', 'Bearer ' + accessToken);
+  headers.set('Pelm-Client-Id', PELM_CLIENT_ID);
+  headers.set('Pelm-Secret', PELM_SECRET);
+
+  const requestOptions = {
+    method: 'GET',
+    headers,
+  };
+
+  const url = 'https://api.pelm.com/bills?account_id=' + req.body.billsAccountIdInput;
+
+  // Reach out to the Pelm API bills endpoint providing all the relevent headers
+  fetch(url, requestOptions)
+    .then((response) => {
+      if (response.ok) {
+        return response.json();
+      } else {
+        return response.text().then((text) => {
+          throw new Error(text);
+        });
+      }
+    })
+    .then((data) => {
+      // Return bills data
+      res.end(JSON.stringify(data));
+    })
+    .catch((error) => {
+      try {
+        const errorObject = JSON.parse(error.message);
+        console.log(errorObject);
+      } catch (e) {
+        console.log('an error occurred');
+      }
+    });
+})
+
+
+// Create access_token
+// Link to Pelm docs: https://docs.pelm.com/reference/post_auth-token-1
+app.post('/authorization', async (req, res) => {
+
+  const headers = new Headers();
+  headers.set('Pelm-Client-Id', PELM_CLIENT_ID);
+  headers.set('Pelm-Secret', PELM_SECRET);
+
+  const encodedParams = new URLSearchParams();
+  encodedParams.set('code', req.body.authorizationCode);
+
+  const requestOptions = {
+    method: 'POST',
+    headers,
+    body: encodedParams,
+  };
+
+  // Reach out to the Pelm API authorization endpoint providing all the relevent headers
+  await fetch('https://api.pelm.com/auth/token', requestOptions)
+    .then((response) => {
+      if (response.ok) {
+        return response.json()
+      } else {
+        return response.text().then((text) => {
+          throw new Error(text);
+        });
+      }
+    })
+    .then(data => {
+      // Save the access_token to your db
+      req.session.access_token = data.access_token
+      // Use the access_token to make requests for a given user's energy data
```

Comment:
suggestions

We recommend securely saving your `access_token` to your database. We're saving to `session` here for convenience.

Reply:
Replaced like 235